forem/forem
# Allow org admins to delete their organizations

## What type of PR is this? (check all applicable)

## Description
This PR allows users to delete their own organizations, if:
- there is only one member in the organization
- they are an organization admin
- the organization has no articles connected to it

While this isn't a heavily requested feature or anything, it's something we should support. Also, this should be one less support request that @michael-tharrington has to handle.

Related to #7798

## QA Instructions, Screenshots, Recordings
![Screen Shot 2020-07-16 at 12 56 13 PM](https://user-images.githubusercontent.com/17884966/87700071-c3452600-c763-11ea-88a8-72cd46159f00.png)
![Screen Shot 2020-07-16 at 1 06 39 PM](https://user-images.githubusercontent.com/17884966/87701010-4155fc80-c765-11ea-8c37-bad7ebd557d1.png)

1. Create an organization
2. Add a user via console:
```ruby
OrganizationMembership.create(user: User.first, organization: Organization.last, type_of_user: "member")
```
3. See that you can't delete it
4. Remove the user (either via settings or via the same script above, except `.destroy`)
5. Try to delete your organization via settings

## Added tests?

## Added to documentation?

## [optional] Are there any post deployment tasks we need to perform?
Nope

## [optional] What gif best describes this PR or how it makes you feel?
!["It's about damn time."](https://media.giphy.com/media/jVStxzak9yk2Q/giphy.gif)
